packages/react-virtualizer/src/components/Virtualizer/Virtualizer.component-browser-spec.tsx

```diff
@@ -0,0 +1,348 @@
+    await expect(component.getByTestId('item-999')).toBeVisible({
+      timeout: 5000,
+    });
+  });
+
+  test('should handle scroll to top', async ({ mount, page }) => {
+    const component = await mount(<VirtualizerExample numItems={1000} />);
+    const container = component.getByTestId('scroll-container');
+
+    // First scroll to middle
+    await container.evaluate((el) => {
+      el.scrollTop = 5000;
+    });
+
+    // Wait for middle items to appear
+    await page.waitForTimeout(500);
+
+    // Then scroll back to top
+    await container.evaluate((el) => {
+      el.scrollTop = 0;
+    });
+
+    // Wait for first item to reappear
+    await expect(component.getByTestId('item-0')).toBeVisible({
+      timeout: 5000,
+    });
+  });
+
+  test('should maintain correct item count', async ({ mount, page }) => {
+    const component = await mount(<VirtualizerExample numItems={1000} />);
+    const container = component.getByTestId('scroll-container');
+
+    await expect(container).toBeVisible();
+
+    // Count rendered items
+    const itemCount = await component.locator('[role="listitem"]').count();
+
+    // Should render more than viewport but less than total
+    expect(itemCount).toBeGreaterThan(5);
+    expect(itemCount).toBeLessThan(1000);
+
+    // After scrolling, count should be similar
+
+    await container.evaluate((el) => {
+      el.scrollTop = 2500;
+    });
+
+    // Wait for virtualization update with longer timeout
+    await page.waitForTimeout(800);
+
+    const itemCountAfterScroll = await component
+      .locator('[role="listitem"]')
+      .count();
+    expect(itemCountAfterScroll).toBeGreaterThan(5);
+    expect(itemCountAfterScroll).toBeLessThan(1000);
+  });
+
+  test('should handle small lists (no virtualization needed)', async ({
+    mount,
+  }) => {
+    const component = await mount(<VirtualizerExample numItems={5} />);
+
+    // All items should be visible
+    await expect(component.getByTestId('item-0')).toBeVisible();
+    await expect(component.getByTestId('item-1')).toBeVisible();
+    await expect(component.getByTestId('item-2')).toBeVisible();
+    await expect(component.getByTestId('item-3')).toBeVisible();
+    await expect(component.getByTestId('item-4')).toBeVisible();
+  });
+
+  test('should render items with correct content', async ({ mount }) => {
+    const component = await mount(<VirtualizerExample numItems={100} />);
+
+    const item0 = component.getByTestId('item-0');
+    await expect(item0).toContainText('Item 0');
+
+    const item1 = component.getByTestId('item-1');
+    await expect(item1).toContainText('Item 1');
+  });
+
+  test('should handle edge case at list boundaries', async ({ mount }) => {
+    const component = await mount(<VirtualizerExample numItems={100} />);
+    const container = component.getByTestId('scroll-container');
+
+    // Scroll to very bottom
+    await container.evaluate((el) => {
+      el.scrollTop = el.scrollHeight - el.clientHeight;
```

Comment:
should we also dispatch a scroll event, wdyt? 

```suggestion
      el.scrollTop = el.scrollHeight - el.clientHeight;\
      el.dispatchEvent(new Event('scroll'));
```